src/nspawn/nspawn.c

```diff
@@ -5279,19 +5293,24 @@ static int run_container(
         }
 
         /* Wait for the outer child. */
-        r = wait_for_terminate_and_check("(sd-namespace)", *pid, WAIT_LOG_ABNORMAL);
+        r = pidref_wait_for_terminate_and_check("(sd-namespace)", pid, WAIT_LOG_ABNORMAL);
```

Comment:
Maybe better to call `pidref_done()` after this get zero or positive return value?